### PR TITLE
DOC: Add Windows build doc

### LINKED ISSUE #4420: Add Windows specific build instructions
Over in #4411 I recommended improving the "Build from source" instructions to break it up into per-platform subsections, including a section for Windows.

However, as I noted in https://github.com/astropy/astropy/pull/4416#issuecomment-167670065 there are not currently _any_ build instructions for Windows.  Only a handful of us, perhaps myself and maybe @mwcraig have direct experience with this and would be able to write such a section.

In particular it would be nice to have instructions for building within the Anaconda environment, and without it.  And separate instructions for setting up and using MSVC or MinGW.  This is a non-trivial task so I think I'll break it out from #4411 which should only add sections for Linux and OSX based on the text that's already there.

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to clarify the process to build `astropy` on Windows.

Fixes #4420

### CODE REVIEW

**Comment by bsipocz** [comment]
@lglattly - could you have a look at this?

File: docs/development/building.rst [hunk 1]
```diff
@@ -25,3 +27,28 @@ of PLY is bundled with Astropy or some of the parsing code changes), the tables
 need to be deleted and the appropriate parts of astropy re-imported and run. For
 exact details, see the comments in the headers of the ``parsetab`` and
 ``lextab`` files.
+
+.. _dev-build-astropy-subpkg-win:
+
+Building on Windows
+*******************
+
+The easiest option is to use Python installation from Miniconda. If you like
```
Comment (lglattly) [inline docs/development/building.rst:36]:
```suggestion
The most convenient option is to use Python installation from Miniconda. If you like
```

File: docs/development/building.rst [hunk 2]
```diff
@@ -25,3 +27,28 @@ of PLY is bundled with Astropy or some of the parsing code changes), the tables
...
+Since ``astropy`` contains C extensions, you also need to install Microsoft
+Visual Studio (the latest available should work) so Python can access the
+system C compiler.
+
+Once everything is set up as above, you can proceed to build ``astropy``
+from source in the ``conda`` environment in an OS-agnostic way. For example:
+
+* Create a new ``conda`` environment.
+* Go to the ``astropy`` code checkout directory.
+* If you have not already, fetch all the tags from the main repository.
```
Comment (lglattly) [inline docs/development/building.rst:51]:
```suggestion
* If you have not already, fetch all of the tags from the main repository.
```

**Comment by pllim** [comment]
p.s. I cancelled everything but the `html-docs` job on CircleCI. FYI.

**Comment by bsipocz** [comment]
Thanks you @pllim! It's so great to see when a super old issue get closed.

**Pull request merged**